packages/medusa/src/loaders/__tests__/models.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { createMedusaContainer } from "medusa-core-utils"
+import path from "path"
+import { asValue } from "awilix"
+
+import modelsLoader from "../models"
+
+describe("models loader", () => {
+  const container = createMedusaContainer()
+  container.register("db_entities", asValue([]))
+  let models
+  let error
+
+  beforeAll(async () => {
+    try {
+      models = await modelsLoader({
+        container,
+        isTest: true,
+        coreTestPathGlob: "../models/{product,product-variant}.ts",
+        rootDirectory: path.join(__dirname, "__fixtures__/customizations"),
+        extensionPathGlob: "models/{product,product-variant}.ts",
+      })
+    } catch (e) {
+      error = e
+    }
+  })
+
+  it("error should be falsy & register 2 models", () => {
+    expect(error).toBeFalsy()
+    expect(models).toHaveLength(2)
+  })
+
+  it("ensure that the product model is an extended model", () => {
+    const productModel = models.find((model) => model.name === "Product")
+
+    expect(productModel.isExtendedModel).toBe(true)
```

Comment:
nit: Seemingly unnecessary, as the assertion on the `custom_attribute` assures that it is the extended model no? This property would not exist on extended models in a real-life scenario
```suggestion
```